# Proyecto: Precios de Laptos parte 2

In [92]:
# Módulos PANDAS y NUMPY
import pandas as pd
import numpy as np


In [95]:
# Jalar el data frame
datos=pd.read_csv("laptopPrice.csv")
# Ojo con las diagonales cuando tengan que definir otra ruta..
# pd.read_csv("C:/Users/migue/Documents/Diplomado_ciencia_datos/datos/nombre_archivo.csv")

In [96]:
datos.head()

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0


In [101]:
datos.shape

(823, 19)

In [100]:
datos.columns

Index(['brand', 'processor_brand', 'processor_name', 'processor_gnrtn',
       'ram_gb', 'ram_type', 'ssd', 'hdd', 'os', 'os_bit', 'graphic_card_gb',
       'weight', 'warranty', 'Touchscreen', 'msoffice', 'Price', 'rating',
       'Number of Ratings', 'Number of Reviews'],
      dtype='object')

## Renombrar columnas de un DataFrame
Nos apoyaremos de un diccionario que nos permita hacer la traducción

In [164]:
nombres={'brand':"marca", 'processor_brand':"marca_procesador", 'processor_name':"nombre_procesador",
         'processor_gnrtn':"generación_procesador",'ram_gb':"ram", 'ram_type':"tipo_memoria", 
         'ssd':"sss", 'hdd':"hdd", 'os':"sistema_operativo", 'os_bit':"bits_so",
         'graphic_card_gb':"tarjeta_grafica",'weight':"peso", 'warranty':"garantía", 
         'Touchscreen':"pantalla_táctil", 'msoffice':"MS_office", 'Price':"Precio", 'rating':"Puntaje",
         'Number of Ratings':"Num_puntuaciones", 'Number of Reviews':"Num_revisiones"}

In [165]:
datos.head()

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,Muy barata
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,Muy barata
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,Muy barata
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0,Cara
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,Muy barata


In [166]:
datos.rename(columns=nombres)  # Regresa una copia del dataframe
# Si se desea que los cambios se apliquen al DataFrame, utilizamos
# el parámetro inplace..
datos.rename(columns=nombres,inplace=True)

In [167]:
datos.head()

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,Muy barata
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,Muy barata
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,Muy barata
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0,Cara
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,Muy barata


In [111]:
# Cómo seleccionamos algunos atributos/columnas/series del dataframe

datos["marca"].head()   # Seleccionamos la serie/columna relativa a la marca



0      ASUS
1    Lenovo
2    Lenovo
3      ASUS
4      ASUS
Name: marca, dtype: object

In [114]:
# Si deseamos seleccionar más de dos atributos
datos_reducido=datos[["marca","ram","Precio","garantía"]]


In [116]:
datos_reducido.head()

,marca,ram,Precio,garantía
0,ASUS,4 GB,34649,No warranty
1,Lenovo,4 GB,38999,No warranty
2,Lenovo,4 GB,39999,No warranty
3,ASUS,8 GB,69990,No warranty
4,ASUS,4 GB,26990,No warranty


In [117]:
# Tipo de garantía que se ofrecen en laptop
datos.garantía.unique()

array(['No warranty', '1 year', '2 years', '3 years'], dtype=object)

In [120]:
#¿Cuál es el tipo de garantía que se adquiere con mayor frecuencia?
datos["garantía"].value_counts(normalize=True)*100


1 year         61.846902
No warranty    33.778858
2 years         2.794654
3 years         1.579587
Name: garantía, dtype: float64

El 61.8% de las laptop que se venden se adquieren con una garantía de un
año, 33.77% no adquieren ningún tipo de garantía, 2.8% de las laptops
se compran con una garantía de hasta dos años y solo un 1.5% adquieren 
una garantía de 3 años.

In [122]:
# Cómo investigamos el precio promedio de las laptops por tipo
# de sistema operativo.
datos["sistema_operativo"].unique()

#Identificamos tres tipos de sistemas operativos. Mac, Windows y DOS 

array(['Windows', 'DOS', 'Mac'], dtype=object)

In [126]:
# Como seleccionamos a las laptops con Windows
datos[datos["sistema_operativo"]=="Windows"].head()

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0


In [128]:
#¿Cómo seleccionamos ahora el precio?
type(datos[datos["sistema_operativo"]=="Windows"])
# Es un DataFrame

pandas.core.frame.DataFrame

In [133]:
datos[datos["sistema_operativo"]=="Windows"]["Precio"].mean()

73125.13863928113

In [134]:
datos[datos["sistema_operativo"]=="Mac"]["Precio"].mean()

151707.85714285713

In [135]:
datos[datos["sistema_operativo"]=="DOS"]["Precio"].mean()

121811.125

In [136]:
# Todo lo anterior lo podemos poner esto en un Loop for.
for so in datos["sistema_operativo"].unique():
    print("""El precio promedio de una laptop con sistema operativo {}, es
          de {:.2f} +/- {:.2f} dólares""".format(so,
          datos[datos["sistema_operativo"]==so]["Precio"].mean(),
          datos[datos["sistema_operativo"]==so]["Precio"].std()))

El precio promedio de una laptop con sistema operativo Windows, es
          de 73125.14 +/- 41344.74 dólares
El precio promedio de una laptop con sistema operativo DOS, es
          de 121811.12 +/- 63633.73 dólares
El precio promedio de una laptop con sistema operativo Mac, es
          de 151707.86 +/- 54907.26 dólares


print("""Los tipos de sistema operativo que se manejan en las laptops influye 
      en el costo final de las mismas, este costo final impacta
      en la decisión del consumidor""")

### Recodificación de una variable, crear una variable a partir de otra

Definimos una computadora "barata" si tiene un Precio es menor a los 50,000
y decimos que es "cara" en otro caso. 


Utilizamos el método where de Numpy...
sintaxis:

np.where(condición, valor en caso de condición==True,valor en caso de condición==False)

IMPORTANTE: Solo recodificación binaria

In [141]:
np.where(datos["Precio"]<=50000,"barata","cara")
# Cómo agraegamos estos resultados al DataFrame
datos["Tipo_laptop"]=np.where(datos["Precio"]<=50000,"barata","cara")

In [142]:
datos.head()

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,barata
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,barata
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,barata
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0,cara
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,barata


In [143]:
# Recodificación múltiple...
datos["Precio"].describe()

count       823.000000
mean      76745.177400
std       45101.790525
min       16990.000000
25%       46095.000000
50%       64990.000000
75%       89636.000000
max      441990.000000
Name: Precio, dtype: float64

Para una clasificación múltiple utilizamos el método select de Numpy.
condiciones=[condicion1, condicion2, ..., condicionesN]
valores=[valor1,valor2,....,valorN]

In [152]:
# NOTA: En pandas la "y" la intersección se denota por &
#                 la "o" la unión se denota por |
condiciones=[datos["Precio"]<=46095,
             (datos["Precio"]>46095)&(datos["Precio"]<=64990),
             (datos["Precio"]>64990)&(datos["Precio"]<=89636),
             datos["Precio"]>=89636]
valores=["Muy barata","Barata", "Cara", "Muy Cara"]
datos["Tipo_laptop"]=np.select(condiciones,valores)

In [153]:
datos.head

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,Muy barata
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,Muy barata
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,Muy barata
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0,Cara
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,Muy barata
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0,Muy barata
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,21990,3 stars,31,3,Muy barata
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,58799,3 stars,0,0,Barata
8,Lenovo,Intel,Core i5,10th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,49999,3 stars,0,0,Barata
9,acer,AMD,Ryzen 5,10th,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,4 GB,Casual,No warranty,No,No,59990,4 stars,1946,240,Barata


# Filtrado de un dataFrame (continuación)

In [155]:
#Seleccionar las laptops marca DELL
datos[datos["marca"]=="DELL"]


,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
29,DELL,Intel,Core i3,10th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39490,4 stars,1267,171,Muy barata
36,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41990,3 stars,10,5,Muy barata
39,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,42490,4 stars,9,1,Muy barata
40,DELL,Intel,Core i3,10th,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,36990,4 stars,1178,137,Muy barata
44,DELL,Intel,Core i3,10th,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,Yes,39490,4 stars,634,64,Muy barata
47,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,No,42990,4 stars,17,2,Muy barata
48,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,49990,4 stars,24,5,Barata
49,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41490,4 stars,1753,165,Muy barata
50,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,41990,4 stars,216,21,Muy barata
59,DELL,Intel,Core i3,11th,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,44427,3 stars,0,0,Muy barata


In [156]:
#Seleccionar las laptops marca DELL y HP
datos[(datos["marca"]=="DELL")|(datos["marca"]=="HP")]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,os_bit,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
15,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,56490,3 stars,0,0,Barata
16,HP,Intel,Core i5,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65390,3 stars,0,0,Cara
18,HP,Intel,Celeron Dual,Not Available,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,32490,4 stars,58,6,Muy barata
22,HP,Intel,Core i3,7th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,32955,4 stars,311,27,Muy barata
23,HP,Intel,Core i5,7th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,46200,4 stars,47,7,Barata
28,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39990,4 stars,273,30,Muy barata
29,DELL,Intel,Core i3,10th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39490,4 stars,1267,171,Muy barata
30,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,Yes,53990,4 stars,14,0,Barata
36,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41990,3 stars,10,5,Muy barata
39,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,42490,4 stars,9,1,Muy barata


In [ ]:
3+5+6

In [171]:
#Seleccionar las laptops marca DELL y HP con sistema operativo Windows
datos[((datos["marca"]=="DELL")|(datos["marca"]=="HP"))&(datos["sistema_operativo"]=="Windows")]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
15,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,56490,3 stars,0,0,Barata
16,HP,Intel,Core i5,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65390,3 stars,0,0,Cara
18,HP,Intel,Celeron Dual,Not Available,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,32490,4 stars,58,6,Muy barata
22,HP,Intel,Core i3,7th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,32955,4 stars,311,27,Muy barata
23,HP,Intel,Core i5,7th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,46200,4 stars,47,7,Barata
28,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39990,4 stars,273,30,Muy barata
29,DELL,Intel,Core i3,10th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39490,4 stars,1267,171,Muy barata
30,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,Yes,53990,4 stars,14,0,Barata
36,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41990,3 stars,10,5,Muy barata
39,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,42490,4 stars,9,1,Muy barata


In [169]:
datos[(datos["sistema_operativo"]=="Windows")&(datos["bits_so"]=="64-bit")]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,Muy barata
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,Muy barata
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,Muy barata
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,Muy barata
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0,Muy barata
12,Avita,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,17490,3 stars,1120,196,Muy barata
13,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0,Muy barata
14,ASUS,Intel,Core i3,11th,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,35990,3 stars,0,0,Muy barata
15,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,56490,3 stars,0,0,Barata
16,HP,Intel,Core i5,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65390,3 stars,0,0,Cara


In [172]:
# Otra manera: Seleccionar laptops HP o DELL con Sistema Operativo Windows
filtro1 = (datos["marca"]=="DELL")
filtro2 = (datos["marca"]=="HP")
filtro3 = (datos["sistema_operativo"]=="Windows")

datos[(filtro1|filtro2)&filtro3]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
15,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,56490,3 stars,0,0,Barata
16,HP,Intel,Core i5,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65390,3 stars,0,0,Cara
18,HP,Intel,Celeron Dual,Not Available,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,32490,4 stars,58,6,Muy barata
22,HP,Intel,Core i3,7th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,32955,4 stars,311,27,Muy barata
23,HP,Intel,Core i5,7th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,46200,4 stars,47,7,Barata
28,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39990,4 stars,273,30,Muy barata
29,DELL,Intel,Core i3,10th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39490,4 stars,1267,171,Muy barata
30,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,Yes,53990,4 stars,14,0,Barata
36,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41990,3 stars,10,5,Muy barata
39,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,42490,4 stars,9,1,Muy barata


In [174]:
# Si el filtro se aplica en un solo atributo...
# Podemos emplear el método de Pandas llamado isin()
# Sintaxis:
# nombre_data_frame["atributo"].isin([lista])


# Ejemplo seleccionar las computadoras DELL, HP, MAC

datos[datos["marca"].isin(["DELL","HP","MAC"])]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
15,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,No,56490,3 stars,0,0,Barata
16,HP,Intel,Core i5,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65390,3 stars,0,0,Cara
18,HP,Intel,Celeron Dual,Not Available,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,32490,4 stars,58,6,Muy barata
22,HP,Intel,Core i3,7th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,32955,4 stars,311,27,Muy barata
23,HP,Intel,Core i5,7th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,46200,4 stars,47,7,Barata
28,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39990,4 stars,273,30,Muy barata
29,DELL,Intel,Core i3,10th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,39490,4 stars,1267,171,Muy barata
30,HP,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,Yes,Yes,53990,4 stars,14,0,Barata
36,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,41990,3 stars,10,5,Muy barata
39,DELL,Intel,Core i3,11th,8 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,42490,4 stars,9,1,Muy barata


In [176]:
datos[datos["ram"].isin(["16 GB"])]

,marca,marca_procesador,nombre_procesador,generación_procesador,ram,tipo_memoria,sss,hdd,sistema_operativo,bits_so,tarjeta_grafica,peso,garantía,pantalla_táctil,MS_office,Precio,Puntaje,Num_puntuaciones,Num_revisiones,Tipo_laptop
218,HP,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,67990,4 stars,108,6,Cara
220,Lenovo,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,62300,4 stars,825,95,Barata
223,Lenovo,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,61990,4 stars,56,7,Barata
232,MSI,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,4 GB,Casual,2 years,No,No,76990,4 stars,75,17,Cara
233,ASUS,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,6 GB,Gaming,1 year,No,Yes,89990,4 stars,124,17,Muy Cara
238,Lenovo,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,64990,3 stars,0,0,Barata
239,ASUS,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,65990,3 stars,0,0,Cara
241,ASUS,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,73490,4 stars,213,45,Cara
242,HP,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,1 year,No,Yes,74490,4 stars,10,0,Cara
245,DELL,Intel,Core i5,11th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,0 GB,ThinNlight,No warranty,No,No,67990,4 stars,93,9,Cara
